tests/unit-tests/internal/emails/test-class-email-customization.php

```diff
@@ -21,4 +21,44 @@ public function testInstance_WhenCalled_ReturnsInstance() {
 		/* Assert. */
 		$this->assertInstanceOf( Email_Customization::class, $result );
 	}
+
+	public function testInstace_WhenInitiated_AddsHookForRemovingLegacyEmail() {
+		/* Arrange. */
+		$settings = $this->createMock( Sensei_Settings::class );
+		$instance = Email_Customization::instance( $settings );
+
+		/* Act. */
+		$instance->init();
+
+		/* Assert. */
+		$this->assertEquals( 10, has_action( 'init', [ $instance, 'disable_legacy_emails' ] ) );
+	}
+
+	/**
+	 * Tests that the legacy email hooks are removed when the disable_legacy_emails function is called.
+	 *
+	 * @param string $action_name The name of the action.
+	 * @param string $function_name The name of the function.
+	 *
+	 * @dataProvider legacyHooksDataProvider
+	 */
+	public function testDisableLegacy_WhenCalled_RemovesLegacyEmailHooks( $action_name, $function_name ) {
+		/* Arrange. */
+		$settings        = $this->createMock( Sensei_Settings::class );
+		$instance        = Email_Customization::instance( $settings );
+		$priority_before = has_action( $action_name, [ \Sensei()->emails, $function_name ] );
```

Comment:
```suggestion
```
We don't need to keep it in case we just [check that the function is not registered.](https://github.com/Automattic/sensei/pull/6567/files#r1118711426)